### PR TITLE
Issue #288 Moved mui-root stylings in various css files to src/assets…

### PR DESCRIPTION
**Pull request by Bugbuster2004**
**Description**:
This pull request addresses issue **#288** by consolidating _MUI-specific component styles from various CSS_ files into the centralized _theme.jsx_ file. This approach ensures better maintainability, consistency, and scalability across the application’s design system.

**Key Changes Made:**

_Migration of Styles:_
Moved the **MuiTab-root styles** from _src/components_ to the _theme.jsx_ file under the overrides section.
Reviewed and migrated similar MUI-related component styles from other CSS files to the centralized theme configuration.
Theme Customization:

Added custom **typography configurations such as font-family, font-size, font-style, font-weight, and line-height for MuiTab-root** and other applicable components in the theme.
Ensured that the styles align with the Inter font family as per the design requirements.

Code Cleanup:
Removed redundant CSS files or declarations for MUI components that are now covered under the _theme.jsx_ configuration.
Refactored the CSS file for HeadingTabs to retain only non-MUI-specific stylings.

Branch Specifics:
All changes have been made on the develop branch, which is the most up-to-date branch for this project.

### CODE REVIEW

File: frontend/src/assets/theme.jsx [hunk 1]
```diff
@@ -12,23 +12,238 @@ export const lightTheme = createTheme({
...
+    //MuiSvgIcon
+    MuiSvgIcon: {
       styleOverrides: {
         root: {
-          backgroundColor: "white",
-          color: "#344054",
+          color: "var(--main-text-color)",
         },
       },
     },
```
[the remark this answers is not in the document]
Comment (Bugbuster2004) [reply frontend/src/assets/theme.jsx:22]: i have applied the changes suggested by you

File: frontend/src/assets/theme.jsx [hunk 2]
```diff
@@ -12,23 +12,231 @@ export const lightTheme = createTheme({
...
+        },
+      },
+    },
+    // MuiListItem Styles (dropdown menu items)
+    MuiListItem: {
+      styleOverrides: {
+        root: {
+          cursor: "pointer",
+          "&:hover": {
+            backgroundColor: "#F9FAFB",
```
[the remark this answers is not in the document]
Comment (Bugbuster2004) [reply frontend/src/assets/theme.jsx:76]: i have made the css variables changes as suggested just wanted to confirm do i need to do the same for borders also?
image

File: frontend/src/index.css [hunk 3]
```diff
@@ -1,4 +1,25 @@
-@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap');
+@import url("https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap");
+
+/* Defining CSS Variables */
+:root {
+  --color-black: #000;
```
Comment (erenfn) [inline frontend/src/index.css:5]: Same variables are also in `variables.css`
Is it possible to get rid of one of them?

**Review by erenfn — CHANGES_REQUESTED** [review]
Can you also remove related stylings from other css files as mui-root is now handled here?

File: frontend/src/index.css (hunk 3 above)
Comment (Bugbuster2004) [reply frontend/src/index.css:5]: we can do one thing that 
image
we can shift these variable color to variable.css and remove it from index.css.
Kindly Confirm if i am thinking right , open to suggestions

Comment (Bugbuster2004) [reply frontend/src/index.css]: made all the changes as requested

File: frontend/src/components/Button/ButtonStyles.css [hunk 4]
```diff
@@ -1,100 +0,0 @@
-
-.button.MuiButton-root{
-  border-radius: 4px;
-  text-transform: none;
-  box-shadow: none;
-}
-
-/* .primary */
-.primary.MuiButton-root {
```
Comment (erenfn) [inline frontend/src/components/Button/ButtonStyles.css:9]: You may have understood the task wrong. You shouldn't delete this file or delete classes like these. `.primary.MuiButton-root` is not the same as `.MuiButton-root`. We are looking for the latter to refactor. all others should stay in place

**Review by erenfn — CHANGES_REQUESTED** [review]

File: frontend/src/components/Button/ButtonStyles.css (hunk 4 above)
Comment (Bugbuster2004) [reply frontend/src/components/Button/ButtonStyles.css:9]: hey @erenfn i undertsood your issue 
but i have to ask that
1. 
image
in this image the primary things is handled using the contained primary 
image

so are you saying that the contained primary logic is extra  and i should remove that or is there a still difference in both

**Review by erenfn — CHANGES_REQUESTED** [review]
It may work, but I cannot test it because this error occurs during the build process:
```
Uncaught Error: MUI: Unsupported `var(--main-purple)` color.
The following formats are supported: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color().
```
Please test your code

**Comment by erenfn** [comment]
@Bugbuster2004 Any updates?

**Comment by Bugbuster2004** [comment]
hey @erenfn  apologies for keeping u waiting , got stuck in some college work , i will work on this tonight and will update you by tomorrow morning. (Indian time)

**Comment by erenfn** [comment]
Hey, that's fine. Thanks for letting me know

**Comment by Bugbuster2004** [comment]
hey @erenfn  i have made the changes 
image
this is the build status
should i send the PR or are there changes to be suggested like regarding these warnings

**Comment by erenfn** [comment]
You can disregard the warnings and send the PR

**Comment by Bugbuster2004** [comment]
Hello @erenfn 

I have raised the PR; however, I was unable to test the frontend code as the backend was not functioning as expected due to some issues with the Docker setup. Though the build process completed successfully without any errors.  

If needed, please let me know, and I would appreciate your guidance in resolving the backend setup process.  

Thank you!

**Review by erenfn — CHANGES_REQUESTED** [review]
The colors don't work correctly. 

image

The things we defined on the css files are all lost

image

Maybe it is just better to stick to the original issue and just refactor the mui-root styles that doesn't have an extra class

**Comment by Bugbuster2004** [comment]
hey @erenfn i got your point , i have to refactor the code again by sticking to the original issue it will take me some days to do that as i have my exams going on currently.

**Comment by erenfn** [comment]
Hey, it has taken a while, so we solved it with #430 . Thanks for your time!

**Pull request closed without merging**